easygems/healpix/__init__.py

```diff
@@ -84,17 +85,34 @@ def healpix_resample(var, xlims, ylims, nx, ny, src_crs, method="nearest", nest=
     res = np.full(latlon.shape[:-1], np.nan, dtype=var.dtype)
 
     if method == "nearest":
-        pix = healpy.ang2pix(
-            healpy.npix2nside(len(var)),
+        pix = healpix.ang2pix(
+            healpix.npix2nside(len(var)),
             theta=points[0],
             phi=points[1],
             nest=nest,
             lonlat=True,
         )
         res[valid] = var[pix]
     elif method == "linear":
-        res[valid] = healpy.get_interp_val(
-            np.asanyarray(var), theta=points[0], phi=points[1], nest=True, lonlat=True
+        lons, lats = healpix.pix2ang(
+            nside=healpix.npix2nside(len(var)),
+            ipix=np.arange(len(var)),
+            nest=nest,
+            lonlat=True,
+        )
+        lons = (lons + 180) % 360 - 180
+
+        valid_src = ((lons > points[0].min()) & (lons < points[0].max())) | (
+            (lats > points[1].min()) & (lats < points[1].max())
+        )
```

Comment:
This will only select points within the bounding box of the output. For a good linear interpolation at the boundary of the plot, we should include one ring of cells outside the view. However, this again is not as straight forward as we'd like using the current state of `healpix`.

Reply:
That's true. But, as you said, there is no straight-forward solution and for the practical tests I did it is not really visible. But of course it will show up if the number of source cells in the region of interest decreases